sw/airborne/modules/rot_wing_drone/rot_wing_automation.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2023 Dennis van Wijngaarden <D.C.vanWijngaarden@tudelft.nl>
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+/** @file "modules/rot_wing_drone/rot_wing_automation.c"
+ * @author Dennis van Wijngaarden <D.C.vanWijngaarden@tudelft.nl>
+ * Fucntions to automate the navigation and guidance of the rotating wing drone
+ */
+
+#include "modules/rot_wing_drone/rot_wing_automation.h"
+#include "state.h"
+#include "modules/datalink/telemetry.h"
+#include "filters/low_pass_filter.h"
+#include "state.h"
+
+#include "math/pprz_algebra_float.h"
+#include "modules/nav/waypoints.h"
+
+#ifndef ROT_WING_AUTOMATION_TRANS_ACCEL
+#define ROT_WING_AUTOMATION_TRANS_ACCEL 1.0
+#endif
+
+#ifndef ROT_WING_AUTOMATION_TRANS_DECEL
+#define ROT_WING_AUTOMATION_TRANS_DECEL 0.5
+#endif
+
+#ifndef ROT_WING_AUTOMATION_TRANS_LENGTH
+#define ROT_WING_AUTOMATION_TRANS_LENGTH 200.0
+#endif
+
+#ifndef ROT_WING_AUTOMATION_TRANS_AIRSPEED
+#define ROT_WING_AUTOMATION_TRANS_AIRSPEED 15.0
+#endif
+
+#ifndef ROT_WING_AUTOMATION_BASE_DIST
+#define ROT_WING_AUTOMATION_BASE_DIST 100.
+#endif
+
+#ifndef ROT_WING_AUTOMATION_WIND_FILT_CUTOFF
+#define ROT_WING_AUTOMATION_WIND_FILT_CUTOFF 0.001
+#endif
+
+struct rot_wing_automation rot_wing_a;
+
+// Declare filters (for windspeed estimation)
+Butterworth2LowPass rot_wing_automation_wind_filter[2]; // Wind filter
+
+// declare function
+inline void update_waypoint_rot_wing_automation(uint8_t wp_id, struct FloatVect3 * target_ned);
+inline void update_wind_vector(void);
+
+void init_rot_wing_automation(void)
+{
+  rot_wing_a.trans_accel = ROT_WING_AUTOMATION_TRANS_ACCEL;
+  rot_wing_a.trans_decel = ROT_WING_AUTOMATION_TRANS_DECEL;
+  rot_wing_a.trans_length = ROT_WING_AUTOMATION_TRANS_LENGTH;
+  rot_wing_a.trans_airspeed = ROT_WING_AUTOMATION_TRANS_AIRSPEED;
+
+  rot_wing_a.transitioned = false;
+  rot_wing_a.windvect.x = 0.0;
+  rot_wing_a.windvect.y = 0.0;
+  rot_wing_a.windvect_f.x = 0.0;
+  rot_wing_a.windvect_f.y = 0.0;
+
+  // Init windvector low pass filter
+  float tau = 1.0 / (2.0 * M_PI * ROT_WING_AUTOMATION_WIND_FILT_CUTOFF);
+  float sample_time = 1.0 / 10.0;
```

Comment:
Frequency would be better to get out of the generated define

Reply:
done